File: merge_sort.js
//javascriptµÝ¹é°æ±¾
Array.prototype.mergeSort = function(){
    var merge = function(left,right){
        var merged = [];
        while(left.length && right.length){
            merged.push(left[0] <= right[0] ? left.shift():right.shift());
        }
        return merged.concat(left.concat(right));
    }
    if (this.length < 2) return this;
    var mid = Math.floor(this.length/2);
    return merge(this.slice(0,mid).mergeSort(),this.slice(mid).mergeSort())
}

File: merge_sort.py
#pythonµÝ¹é°æ±¾
def merge(left,right):
    merged = []
    while left and right:
        merged.append(left.pop(0) if left[0] <= right[0] else right.pop(0))
    while left:
        merged.append(left.pop(0))
    while right:
        merged.append(right.pop(0))
    return merged
    
def merge_sort(a):
    n = len(a)
    if n<= 1:
        return a
    mid = n//2
    return merge(merge_sort(a[:mid]),merge_sort(a[mid:]))